course/src/main/java/org/openedx/course/presentation/container/CourseContainerFragment.kt

```diff
@@ -387,153 +393,178 @@ fun CourseDashboard(
                 }
             }
             HandleUIMessage(uiMessage = uiMessage, scaffoldState = scaffoldState)
+            if (dataReady.value.isNull()) return@Scaffold
+
+            if (dataReady.value.isTrue() && canShowTrackSelection) {
```

Comment:
use-case of adding the `dataReady.value.isTrue()` in the condition, cuz seems `canShowTrackSelection` is not dependent on `dataReady`.